src/oemof/tabular/datapackage/reading.py

```diff
@@ -433,15 +434,108 @@ def create_periodic_values(values, periods_index):
 
         return periodic_values.tolist()
 
-    def create_yearly_values(values, years):
+    def create_yearly_values(
+        values: typing.Iterable[float], period_years: typing.Iterable[int]
+    ):
+        """
+        Creates a value for every year (between two periods)
+        Value of period is continued until next period
+        Parameters
+        ----------
+        values values to be interpolated
+        years years of periods
+
+        Returns list
+        -------
+
+        """
         results = pd.Series()
-        for i in range(len(years) - 1):
-            diff = years[i + 1] - years[i]
+        for i in range(len(period_years) - 1):
+            diff = period_years[i + 1] - period_years[i]
             period_results = pd.Series(repeat(values[i], diff))
             results = pd.concat([results, period_results])
         results = pd.concat([results, pd.Series(values[-1])])
         return results.tolist()
 
+    def convert_and_deserialize_dtypes(facade, period_data):
```

Comment:
How about `convert_sequences()` ? I am not sure about the `dtypes` thing here as the type of sequences is mainly chosen by `parameter-name` isnt it?